internal/pkg/client/gitlab/fetch.go

```diff
@@ -0,0 +1,227 @@
+// -----------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl a Series of LF Projects, LLC
+// SPDX-FileName: internal/pkg/client/gitlab/fetch.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// -----------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+package gitlab
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/xanzy/go-gitlab"
+
+	bomctloptions "github.com/bomctl/bomctl/internal/pkg/options"
+)
+
+type ClientWrapperInterface interface {
+	GetProject(
+		any,
+		*gitlab.GetProjectOptions,
+		...gitlab.RequestOptionFunc,
+	) (*gitlab.Project, *gitlab.Response, error)
+	GetBranch(any, string, ...gitlab.RequestOptionFunc) (*gitlab.Branch, *gitlab.Response, error)
+	GetCommit(
+		any,
+		string,
+		*gitlab.GetCommitOptions,
+		...gitlab.RequestOptionFunc,
+	) (*gitlab.Commit, *gitlab.Response, error)
+	CreateDependencyListExport(
+		int,
+		*gitlab.CreateDependencyListExportOptions,
+		...gitlab.RequestOptionFunc,
+	) (*gitlab.DependencyListExport, *gitlab.Response, error)
+	GetDependencyListExport(int, ...gitlab.RequestOptionFunc) (*gitlab.DependencyListExport, *gitlab.Response, error)
+	DownloadDependencyListExport(int, ...gitlab.RequestOptionFunc) (io.Reader, *gitlab.Response, error)
+}
+
+type clientWrapper struct {
+	gitlab.ProjectsService
+	gitlab.BranchesService
+	gitlab.CommitsService
+	gitlab.DependencyListExportService
+}
+
+type dependencyListExportSession struct {
+	Client ClientWrapperInterface
+	Export *gitlab.DependencyListExport
+}
+
+var (
+	errInvalidGitLabURL = errors.New("invalid URL for GitLab fetching")
+	errFailedWebRequest = errors.New("web request failed")
+	errForbiddenAccess  = errors.New("you don't have permission to read the dependency list")
```

Comment:
Avoid "you" statements

```suggestion
	errForbiddenAccess  = errors.New("the supplied token is missing the read_dependency permission")
```